### PR TITLE
Added codesnippet for sync client and builder

### PR DESCRIPTION
**Pull request by mssfang**
- Fixed sync #6874,
- removed detectLanguage(string text, string countryHint) method to align with other API.
- renamed analyzeBatchSentimentWithResponse(String text, String language, Context context)) to analyzeSentimentWithResponse
- fixed tests, that include added tests for faultyText in sync and async client.

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java [hunk 1]
```diff
@@ -51,35 +51,27 @@
...
      *
      * @param text The text to be analyzed.
-     * @param countryHint Accepts two letter country codes specified by ISO 3166-1 alpha-2. Defaults to "US" if not
-     * specified.
-     *
      * @return the {@link DetectLanguageResult detected language} of the text.
      * @throws NullPointerException if {@code text} is {@code null}.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public DetectLanguageResult detectLanguage(String text, String countryHint) {
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java]: why is the method getting changed here?

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java [hunk 2]
```diff
@@ -51,35 +51,27 @@
...
-     */
-    @ServiceMethod(returns = ReturnType.SINGLE)
-    public DetectLanguageResult detectLanguage(String text) {
-        return detectLanguage(text, client.getDefaultCountryHint());
-    }
-
-    /**
-     * Returns the detected language and a numeric score between zero and one when the hint of country specified.
-     * Scores close to one indicate 100% certainty that the identified language is true.
+     * <p><strong>Code Samples</strong></p>
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java]: Code Sample, here and rest of the file.

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java [hunk 3]
```diff
@@ -46,6 +46,14 @@
  * {@link #credential(TokenCredential) credential(TokenCredential)} give the builder access credential.
  * </p>
  *
+ * <p><strong>Instantiating an asynchronous Text Analytics Client</strong></p>
+ *
+ * {@codesnippet com.azure.ai.textanalytics.TextAnalyticsAsyncClient.instantiation}
+ *
+ * <p><strong>Instantiating a synchronous Text Analytics Client</strong></p>
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java]: twice?

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/TextAnalyticsClientJavaDocCodeSnippets.java [hunk 4]
```diff
@@ -3,25 +3,1133 @@
...
+
+    /**
+     * Code snippet for {@link TextAnalyticsClient#detectLanguagesWithResponse(List, String, Context)}
+     */
+    public void detectLanguageForListInputTextsWithResponse() {
+        // BEGIN: com.azure.ai.textanalytics.TextAnalyticsClient.detectLanguagesWithResponse#List-String-Context
+        final List<String> textInputs = Arrays.asList(
+            "This is written in English",
+            "Este es un document escrito en Español.");
+        final DocumentResultCollection<DetectLanguageResult> detectLanguageResults =
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/TextAnalyticsClientJavaDocCodeSnippets.java]: nit: can remove final 
here and other examples.

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/TextAnalyticsClientJavaDocCodeSnippets.java [hunk 5]
```diff
@@ -3,25 +3,1133 @@
...
+    public void recognizeEntitiesSingleText() {
+        // BEGIN: com.azure.ai.textanalytics.TextAnalyticsClient.recognizeEntities#String
+        final RecognizeEntitiesResult recognizeEntitiesResult =
+            textAnalyticsClient.recognizeEntities("Satya Nadella is the CEO of Microsoft");
+        for (NamedEntity entity : recognizeEntitiesResult.getNamedEntities()) {
+            System.out.printf(
+                "Recognized entity: %s, entity type: %s, entity subtype: %s, offset: %s, length: %s, score: %s.%n",
+                entity.getText(),
+                entity.getType(),
+                entity.getSubtype() == null || entity.getSubtype().isEmpty() ? "N/A" : entity.getSubtype(),
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/TextAnalyticsClientJavaDocCodeSnippets.java]: Can make the snippet shorter?
here and other examples.

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java (hunk 1 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java]: I think this API is not required. It only takes input text and country hint. If you take a look at other APIs. You will find this API is an extra API. Either of AsyncClient doesn't have it or recognizeEntiteis, PII entities, LInked Entities, Extract KeyPhrases and Analyze Sentiments don't have it either.

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java (hunk 3 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java]: one is for Async and another is for Sync. By comparing the length, you will see it intuitively.

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java (hunk 2 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java]: I don't get this?      
```
 <p><strong>Code Samples</strong></p>
```

Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java]: should be singular

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/TextAnalyticsClientJavaDocCodeSnippets.java [hunk 6]
```diff
@@ -3,25 +3,1133 @@
...
  */
 public class TextAnalyticsClientJavaDocCodeSnippets {
     private static final String SUBSCRIPTION_KEY = null;
     private static final String ENDPOINT = null;
+    private final TextAnalyticsClient textAnalyticsClient = new TextAnalyticsClientBuilder().buildClient();
+
+    /**
+     * Code snippet for creating a {@link TextAnalyticsClient} with pipeline
+     */
+    public void createAsyncTextAnalyticsClientWithPipeline() {
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/TextAnalyticsClientJavaDocCodeSnippets.java]: Is this supposed to be in Async file?

Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/TextAnalyticsClientJavaDocCodeSnippets.java]: sorry. wrong method name, async should be removed

File: sdk/textanalytics/azure-ai-textanalytics/README.md [hunk 7]
```diff
@@ -361,7 +342,25 @@ for (TextSentiment textSentiment : textAnalyticsClient.analyzeSentiment(text).ge
 ```
 
 ## Troubleshooting
-## General
+### General
+Text Analytics clients raise exceptions. For example, if you try to detect the languages of a batch of text with same 
+document IDs, 400 error is return that indicating bad request. In the following code snippet, the error is handled 
+gracefully by catching the exception and display the additional information about the error.
+
+<!-- embedme ./src/samples/java/com/azure/ai/textanalytics/ReadmeSamples.java#L164-L169 -->
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/README.md]: correct the line numbers here.

File: sdk/textanalytics/azure-ai-textanalytics/README.md [hunk 8]
```diff
@@ -361,7 +342,25 @@ for (TextSentiment textSentiment : textAnalyticsClient.analyzeSentiment(text).ge
 ```
 
 ## Troubleshooting
-## General
+### General
+Text Analytics clients raise exceptions. For example, if you try to detect the languages of a batch of text with same 
+document IDs, 400 error is return that indicating bad request. In the following code snippet, the error is handled
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/README.md]: nit: 400 -> `404`

Comment (conniey) [reply sdk/textanalytics/azure-ai-textanalytics/README.md]: 404 is an interesting HTTP error code for "Bad Request'. Traditionally, 400 is bad request... and 404 is "Not found". (ie. this URL you made does not exist).

Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/README.md]: https://en.wikipedia.org/wiki/HTTP_404

Comment (samvaity) [reply sdk/textanalytics/azure-ai-textanalytics/README.md]: oh mistyped I mean 400 itself just syntax highlighted 🙆

**Pull request merged**